runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -6476,343 +6359,174 @@ static void genHeapAlloc(
    }
 
 // ------------------------------------------------------------------------------
-// genHeapAlloc2
+// genHeapAllocForObjectOrHybridArraylet
 //
-// Will eventually become the de facto genHeapAlloc.  Needs packed array and 2TLH
-// support.
+// Needs 2TLH support.
 // ------------------------------------------------------------------------------
 
-static void genHeapAlloc2(
-   TR::Node             *node,
-   TR_OpaqueClassBlock *clazz,
-   int32_t              allocationSizeOrDataOffset,
-   int32_t              elementSize,
-   TR::Register         *sizeReg,
-   TR::Register         *eaxReal,
-   TR::Register         *segmentReg,
-   TR::Register         *tempReg,
-   TR::LabelSymbol       *failLabel,
-   TR::CodeGenerator    *cg)
+/**
+ * @param[in] eaxReal : address of the newly allocated object
+ * @param[in] nextTLHAllocReg : the new TLH alloc pointer after the object is allocated
+ */
```

Comment:
Is it worth clearly stating what the difference is with respect to `genHeapAlloc` and the conditions under which either one gets used ? Just so there is more clarity (in addition to the re-naming and doc you added already, that are all steps in the right direction) ?

Reply:
I will rename `genHeapAlloc` to `genHeapAllocForDiscontiguousArraysOrRealtime`

Reply:
Fixed in latest force push.